### PR TITLE
Moving File#fsync, File#flock_exclusive, File#flock_shared, and File#flock_unlock methods to IO::FileDescriptor - Fixes #9793

### LINKED ISSUE #9793: File#fsync, File#flock_exclusive, File#flock_shared, and File#flock_unlock belong in IO::FileDescriptor
Unix-style file operations, like `File#fsync`, `File#flock_exclusive`, `File#flock_shared`, and `File#flock_unlock` belong in `IO::FileDescriptor`. On Windows these are completely different APIs, which perhaps is why they are presently in `File`. However, this is a significant handicap on Unix-like systems and makes the user want to import the API from `LibC`, which would be ugly and even less portable.

An example of how to code these APIs portably to Windows is at https://github.com/WoLpH/portalocker

Why is the user using `IO::FileDescriptor`? They are doing operations on `STDIN`, `STDOUT`, or `STDERR` (which can be legitimate files), or on a file descriptor inherited from the parent process, or passed in from a foreign library.

### PR DESCRIPTION
**Pull request by naqvis**
Moved `File#fsync, File#flock_exclusive, File#flock_shared, File#flock_unlock` to `IO::FileDescriptor`
https://github.com/crystal-lang/crystal/issues/9793#issuecomment-703103653

### CODE REVIEW

File: src/io/file_descriptor.cr [hunk 1]
```diff
@@ -133,6 +133,60 @@ class IO::FileDescriptor < IO
     value
   end
 
+  # Flushes all data written to this File to the disk device so that
```
Comment (asterite) [inline src/io/file_descriptor.cr:136]: Maybe change File to File Descriptor? Same in other places if it happens.

**Review by naqvis — COMMENTED** [review]
Done. Changes `File` to `File Descriptor` in comments.

File: src/io/file_descriptor.cr (hunk 1 above)
Comment (naqvis) [reply src/io/file_descriptor.cr]: Done. Thank you @asterite

**Comment by oprypin** [comment]
Please change the title of this PR, right now it's basically empty.

**Comment by naqvis** [comment]
> Please change the title of this PR, right now it's basically empty.

Thank you @oprypin. Changed title to match with issue heading

**Review by asterite — APPROVED** [review]

**Review by bcardiff — APPROVED** [review]
Eventually the specs for IO::FileDescriptor should get more populated and check that we can use non files descriptors with these operations.

**Pull request merged**